game-core/src/main/java/games/strategy/engine/data/unit/ability/UnitAbilityFactory.java

```diff
@@ -0,0 +1,456 @@
+                generatePerPlayerAndUnit(
+                    unitTypeList, battlePhaseList, properties, player, unitType));
+
+    createBombardUnitAbilities(unitTypeList, battlePhaseList, properties, player);
+  }
+
+  private static void generatePerPlayerAndUnit(
+      final UnitTypeList unitTypeList,
+      final BattlePhaseList battlePhaseList,
+      final GameProperties properties,
+      final GamePlayer player,
+      final UnitType unitType) {
+    final UnitAttachment unitAttachment = UnitAttachment.get(unitType);
+    if (unitAttachment.getIsAaForCombatOnly()) {
+      createAaUnitAbilities(battlePhaseList, player, unitType);
+    }
+    createUnitAbilities(unitTypeList, battlePhaseList, properties, player, unitType);
+  }
+
+  private static void createAaUnitAbilities(
+      final BattlePhaseList battlePhaseList, final GamePlayer player, final UnitType unitType) {
+    final UnitAttachment unitAttachment = UnitAttachment.get(unitType);
+    final Collection<BattleState.Side> sides = new ArrayList<>();
+    if (unitAttachment.getOffensiveAttackAa(player) > 0 && unitAttachment.getMaxAaAttacks() != 0) {
+      sides.add(BattleState.Side.OFFENSE);
+    }
+    if (unitAttachment.getAttackAa(player) > 0 && unitAttachment.getMaxAaAttacks() != 0) {
+      sides.add(BattleState.Side.DEFENSE);
+    }
+    if (sides.isEmpty()) {
+      return;
+    }
+
+    final CombatUnitAbility ability =
+        addAbility(
+            battlePhaseList,
+            CombatUnitAbility.builder()
+                .name(unitAttachment.getTypeAa())
+                .attachedUnitTypes(List.of(unitType))
+                .diceType(CombatUnitAbility.DiceType.AA)
+                .sides(sides)
+                .round(
+                    unitAttachment.getMaxRoundsAa() == -1
+                        ? Integer.MAX_VALUE
+                        : unitAttachment.getMaxRoundsAa())
+                .targets(aaTargets.get(unitAttachment.getTypeAa()))
+                .returnFire(false)
+                .commitSuicideAfterSuccessfulHit(getCommitSuicideOnHitSides(unitType))
+                .commitSuicide(getCommitSuicideSides(unitType))
+                .build(),
+            player,
+            BattlePhaseList.DEFAULT_AA_PHASE);
+
+    unitAttachment.getWillNotFireIfPresent().stream()
+        .map(
+            unitTypeThatPreventsFiring ->
+                createWillNotFireIfPresentAbilities(unitTypeThatPreventsFiring, ability))
+        .forEach(antiAbility -> addConvertAbility(battlePhaseList, antiAbility, player));
+  }
+
+  private static CombatUnitAbility addAbility(
+      final BattlePhaseList battlePhaseList,
+      final CombatUnitAbility combatUnitAbility,
+      final GamePlayer player,
+      final String phaseName) {
+    final BattlePhase battlePhase =
+        battlePhaseList
+            .getPhase(phaseName)
+            .orElseThrow(
+                () -> new IllegalArgumentException("The phase " + phaseName + " doesn't exist"));
```

Comment:
Printing the list of battle phase names in the exception message could make for a better error message, in case there are typos or some small differences, it'll be easier to spot.